### PR TITLE
Fix documentation bugs, update DG, add ReservationTest, add PPP

### PR DESCRIPTION
**Pull request by mukundrs**
- Fixed trailing whitespaces in documentation files
- Updated DG with information relevant to the latest version of ezFoodie and added documentation for add reservation feature
- Updated UML diagrams with information relevant to the latest version of ezFoodie
- Added test for Reservation model
- Added PPP for myself

### CODE REVIEW

**Review by HolmesJJ — APPROVED** [review]
LGTM

File: docs/team/holmesjj.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ This page serves to document my contributions to the project ezFoodie under NUS
 
 ### Project: ezFoodie
 
-ezFoodie is a desktop application that helps restaurants **keep track of their ever-growing list of members**. 
+ezFoodie is a desktop application that helps restaurants **keep track of their ever-growing list of members**.
```
Comment (Morrow1ndy) [inline docs/team/holmesjj.md:12]: e can be capital letter maybe?

File: docs/team/zzybluebell.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ This page serves to document my contributions to the project ezFoodie under NUS
 
 ### Project: ezFoodie
 
-ezFoodie is a desktop application that helps restaurants **keep track of their ever-growing list of members**. 
+ezFoodie is a desktop application that helps restaurants **keep track of their ever-growing list of members**.
```
Comment (Morrow1ndy) [inline docs/team/zzybluebell.md:12]: same here

File: src/test/java/seedu/address/model/reservation/ReservationTest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        String invalidDateTime = "";
+        assertThrows(IllegalArgumentException.class, () -> new DateTime(invalidDateTime));
+    }
+
+    @Test
+    public void isValidDateTime() {
+        // null dateTime String
+        assertThrows(NullPointerException.class, () -> DateTime.isValidDateTime(null));
+    }
+}
```
Comment (Morrow1ndy) [inline src/test/java/seedu/address/model/reservation/ReservationTest.java:25]: Overall LGTM

**Review by Morrow1ndy — COMMENTED** [review]
Small issue only.

**Pull request merged**